clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -215,6 +215,8 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
     if (FunctionDecl *Callee = e->getDirectCallee()) {
       Callee = Callee->getCanonicalDecl();
+      assert(Callee && "Device function canonical decl must be available");
```

Comment:
Why do you need this assertion? `Callee` is already checked for `nullptr` within `if` statement on 216